src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -1406,8 +1406,8 @@ G1CollectedHeap::G1CollectedHeap() :
   _summary_bytes_used(0),
   _bytes_used_during_gc(0),
   _archive_allocator(NULL),
-  _survivor_evac_stats("Young", YoungPLABSize, PLABWeight),
-  _old_evac_stats("Old", OldPLABSize, PLABWeight),
+  _survivor_evac_stats("Young", YoungPLABSize * ParallelGCThreads, PLABWeight),
```

Comment:
Lgtm, will do some perf measurements and get back to you.
Could you add a comment like the following in the G1EvacStats constructor?

// desired_plab_size_ should be the total PLAB size for all threads.

Reply:
Just one note, if we run with `UseDynamicNumberOfGCThreads` the PLAB size might still be "resized" between GCs if they use different number of threads. This is still an improvement since the old value is wrong and will give a way to small PLAB size. But we might want to look at a solution that will always return a fixed PLAB size with `-ResizePLAB`.

Reply:
I agree and noted that in the CR already. As you mention, it is still an improvement, and with `-XX:+ResizePLAB` G1 will automatically resize the PLABs back to more reasonable if needed (as G1 previously used a way too small PLAB initially).

Giving that typical number or threads correlate somewhat with heap size, and the policy for `-XX:+UseDynamicNumberOfThreads` is fairly conservative anyway, this change will "always" return the (correct) fixed PLAB size with `-XX:-ResizePLAB` now.

Reply:
I.e. if you want (and think it is a good idea), we can do both changes, first the one suggested originally, and this one so that we do not need to review + test twice.

Reply:
I agree, but to be able to do both we need to store the "default size" for the `PLAB` in the `PLABStats` as well, right? What would we otherwise return in the `-ResizePLAB` case.

But storing both the default and desired size in `PLABStats` would be ok with me.